content/tutorial/tutorial.md

```diff
@@ -513,63 +519,64 @@ class Board extends React.Component {
+पिछले कोड उदाहरण में, हमने सुझाव दिया कि `.slice()` ऑपरेटर का इस्तेमाल करके `squares` array की कॉपी बनाये और मौजूदा array का संसोधन करने के बजाये नए array का संसोधन करें। अब हम अपरिवर्तनीयता पर चर्चा करेंगे और सीखने के लिए अपरिवर्तनशीलता क्यों महत्वपूर्ण है।
 
-There are generally two approaches to changing data. The first approach is to *mutate* the data by directly changing the data's values. The second approach is to replace the data with a new copy which has the desired changes.
+डेटा बदलने के लिए आम तौर पर दो दृष्टिकोण होते हैं। पहला तरीका डेटा के मूल्यों को सीधे बदलकर डेटा को *म्यूटेट* करना है। ददूसरा दृश्टिकोण डाटा को उसके नई कॉपी से बदलने का है जिसमें ज़रूरी परिवर्तन हैं।
 
-#### Data Change with Mutation {#data-change-with-mutation}
+#### म्यूटेशन के साथ डेटा बदलें {#data-change-with-mutation}
 ```javascript
 var player = {score: 1, name: 'Jeff'};
 player.score = 2;
-// Now player is {score: 2, name: 'Jeff'}
+// अब खिलाड़ी {score: 2, name: 'Jeff'}  ै 
 ```
 
-#### Data Change without Mutation {#data-change-without-mutation}
+#### म्यूटेशन के बिना डेटा बदलें {#data-change-without-mutation}
 ```javascript
 var player = {score: 1, name: 'Jeff'};
 
 var newPlayer = Object.assign({}, player, {score: 2});
-// Now player is unchanged, but newPlayer is {score: 2, name: 'Jeff'}
+// अब खिलाड़ी अपरिवर्तित है, लेकिन newPlayer {score: 2, name: 'Jeff'} है
 
-// Or if you are using object spread syntax proposal, you can write:
+// या यदि आप ऑब्जेक्ट स्प्रेड सिंटैक्स प्रस्ताव का उपयोग कर रहे हैं, तो आप लिख सकते हैं:
 // var newPlayer = {...player, score: 2};
 ```
 
-The end result is the same but by not mutating (or changing the underlying data) directly, we gain several benefits described below.
+स्टेट सीधे मूतते न करके (या अंतर्निहित डेटा का परिवर्तन करे बिना) हमने सामान अंतिम परिणाम पाया, पर इससे हमें निचे वर्णित कई लाभ प्राप्त हुए
 
-#### Complex Features Become Simple {#complex-features-become-simple}
+#### जटिल विशेषताएँ सरल बनें {#complex-features-become-simple}
 
-Immutability makes complex features much easier to implement. Later in this tutorial, we will implement a "time travel" feature that allows us to review the tic-tac-toe game's history and "jump back" to previous moves. This functionality isn't specific to games -- an ability to undo and redo certain actions is a common requirement in applications. Avoiding direct data mutation lets us keep previous versions of the game's history intact, and reuse them later.
+अपरिवर्तनीयता जटिल सुविधाओं को लागू करने में बहुत आसान बनाती है। इस ट्यूटोरियल में बाद में, हम एक "टाइम ट्रैवल" सुविधा को लागू करेंगे, जो हमें टिक-टैक-टो गेम के इतिहास की समीक्षा करने और पिछली चालों के लिए "वापस कूदने" की अनुमति देगा। यह कार्यक्षमता खेलों के लिए विशिष्ट नहीं है - कुछ कार्यों को पूर्ववत और फिर से करने की क्षमता अनुप्रयोगों में एक सामान्य आवश्यकता है। प्रत्यक्ष डेटा उत्परिवर्तन से बचने से हम खेल के इतिहास के पिछले संस्करणों को बरकरार रख सकते हैं, और बाद में उनका पुन: उपयोग कर सकते हैं।
```

Comment:
जटिल सुविधाओं -> jatil features (in hindi) ?

Reply:
टाइम ट्रैवल" सुविधा same

Reply:
वापस कूदने does not seem relevant

Reply:
यह कार्यक्षमता खेलों के लिए विशिष्ट नहीं है - कुछ कार्यों को पूर्ववत और फिर से करने की क्षमता अनुप्रयोगों में एक सामान्य आवश्यकता है। -> revisit?

Reply:
उत्परिवर्तन -> ??

Reply:
> वापस कूदने does not seem relevant

`piche jaane`?

Reply:
> यह कार्यक्षमता खेलों के लिए विशिष्ट नहीं है - कुछ कार्यों को पूर्ववत और फिर से करने की क्षमता अनुप्रयोगों में एक सामान्य आवश्यकता है। -> revisit?

`यह कार्यक्षमता खेलों के लिए विशिष्ट नहीं है -- अनुप्रयोगों में कुछ कार्यों को पूर्ववत और फिर से करने की क्षमता एक सामान्य बात है` ?

Reply:
> उत्परिवर्तन -> ??

`म्युटेशन` ?

Reply:
> > यह कार्यक्षमता खेलों के लिए विशिष्ट नहीं है - कुछ कार्यों को पूर्ववत और फिर से करने की क्षमता अनुप्रयोगों में एक सामान्य आवश्यकता है। -> revisit?
> 
> `यह कार्यक्षमता खेलों के लिए विशिष्ट नहीं है -- अनुप्रयोगों में कुछ कार्यों को पूर्ववत और फिर से करने की क्षमता एक सामान्य बात है` ?

still feels too complex

Reply:
> > > यह कार्यक्षमता खेलों के लिए विशिष्ट नहीं है - कुछ कार्यों को पूर्ववत और फिर से करने की क्षमता अनुप्रयोगों में एक सामान्य आवश्यकता है। -> revisit?
> > 
> > 
> > `यह कार्यक्षमता खेलों के लिए विशिष्ट नहीं है -- अनुप्रयोगों में कुछ कार्यों को पूर्ववत और फिर से करने की क्षमता एक सामान्य बात है` ?
> 
> still feels too complex

any suggestions?

Reply:
probably translate application as it is instead of anuprayag?

Reply:
`यह कार्यक्षमता खेलों के लिए विशिष्ट नहीं है -- एप्लीकेशन में कुछ कार्यों को पूर्ववत और फिर से करने की क्षमता एक सामान्य बात है`
something like this?

Reply:
yeah, just that it should be applications